debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -249,6 +249,20 @@ private void readOplog(MongoClient primary, MongoPrimary primaryClient, ReplicaS
         }
     }
 
+    private boolean isStartPositionInOplog(BsonTimestamp startTime, MongoCollection<Document> oplog) {
+        final MongoCursor<Document> iterator = oplog.find().sort(new Document("$natural", 1)).iterator();
```

Comment:
Is that sorting step even needed? Or is it some pseudo logic anyways, and "natural" will just return the collection as-is? Can we get an execution plan of the two flavours somehow?

Reply:
Looking at both the explain plan and the MongoDB documentation a bit more closely, it would seem the sort is superfluous.  While it does not change the execution plan, since the oplog is a capped collection, the collection is automatically ordered and the find call should return the results in the collection's order from what I have understood.  I'm re-running the tests to make sure there are no regressions.

Reply:
> the collection is automatically ordered and the find call should return the results in the collection's order

Ok, that's what I would have expected too. So let's get rid of the sort and get this one merged, so that the reporter can test this with today's nightly build if they want to.